src/compiler/comments.ts

```diff
@@ -260,7 +260,21 @@ namespace ts {
             }
         }
 
+        function isJSDocLikeText(text: string, start: number) {

```

Comment:
Can't. They behave meaningfully differently. `isJSDocLikeText` matches anything that starts with `**` and isn't an empty comment. `isJsDocStart` only matches comments starting with `**` but not `***` and doesn't filter empty comments at all.

Reply:
> Why does isJsDocStart check for /***

Honestly no idea. It just seems as if we decided at some point that extra stars (like if, for example, you put line-length star guards around your comment) don't count as jsdoc as far as the parser is concerned. However I definitely think it's still correct to preserve such a comment in declaration emit, hence the difference.

Reply:
I've unified them both under the previous `isJSDocLikeText` implementation, since it seems like the parser definition has just sat around since its introduction three years ago, and the only justification for why `/*** @type {Foo} */` is any less jsdoc than `/** @type {Foo} */` I've found is a remark on usejsdoc.org; but we're typically more lenient than it says (plus, [closure compiler allows it](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F**************************%250A*%2520%2540param%2520%257B%257Bx%253A%2520number%257D%257D%2520name%250A***************************%252F%250Afunction%2520hello(name)%2520%257B%250A%2520%2520alert('Hello%252C%2520'%2520%252B%2520name)%253B%250A%257D%250Ahello('New%2520user')%253B%250A%250A) (you can see the type annotation still getting applied under the advanced mode error)) cc @DanielRosenwasser it makes sense to be more lenient here, right?